### PR TITLE
User identity initialize

### PR DESCRIPTION
**Pull request by annaswims**
## Description of change
in our test we were re-using the same uuid, which could lead to inaccurate test results as the same key was used and re-used in redis. 

Unless we initialize user.identity, it will be nil until identity is saved in redis, which means that user will not use the loa3 validations. 

This is the problem that was alluded to in https://github.com/department-of-veterans-affairs/vets-api/pull/2772

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: app/services/sso_service.rb [hunk 1]
```diff
@@ -72,6 +72,8 @@ def authn_context
 
   def init_new_user(user_identity, existing_user = nil, multifactor_change = false)
     new_user = User.new(user_identity.attributes)
+    new_user.identity = user_identity
```
Comment (annaswims) [inline app/services/sso_service.rb]: I'm struggling to find a way to properly test this, but if you put `user_identity.attributes[:loa][:current] == new_user.identity[:loa][:current]` after line 74 you'll see that it's not always true.

**Comment by annaswims** [comment]
I'm not clear if this would ever create issues for the enduser, but I think this is closer to developer's expectations.

File: app/models/user.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@ class User < Common::RedisStore
   redis_store REDIS_CONFIG['user_b_store']['namespace']
   redis_ttl REDIS_CONFIG['user_b_store']['each_ttl']
   redis_key :uuid
-
+  attr_writer :identity
```
Comment (saneshark) [inline app/models/user.rb]: I'm not sure we want to be able to update UserIdentity from user object, this opens up the possibility for certain things I'm weary of introducing, like modifying UserIdentity attributes after the response is received through the user object.

File: app/services/sso_service.rb (hunk 1 above)
Comment (saneshark) [reply app/services/sso_service.rb]: I'm confused as to why you think this is a problem. The idea is that the new_user is built from the newest set of attributes we've received. It should also only be happening if the latest authn_context results in a attribuets class of IdMe... in other words, multifactor or verification.

**Comment by annaswims** [comment]
Does this part of the description not make sense? "Unless we initialize user.identity, it will be nil until identity is saved in redis, which means that user will not use the loa3 validations."

**Comment by saneshark** [comment]
If a user is going to LOA3, we already know that they are eligible for LOA3 from the user identity that is already persisted. The new user identity being initialized or not shouldn't matter. Are you seeing something to suggest otherwise?

**Pull request merged**